### ISSUE TITLE
Error TensorflowModelTrain with path of dimension 48 instead of 16

### ISSUE THREAD

**Issue by viennois** [remicres/otbtf#88]
HI,
I extract patches of dimension 48 instead of 16
Here, my line of code : 
"otbcli_PatchesExtraction -source1.il Mosa_8B_Projet_Pheno_24022022_hillshade.tif -source1.patchsizex 48 -source1.patchsizey 48 -source1.out train_mosafinale17classes_Patches48.tif -vec Training_17classes02032022.shp -field "Num_class" -outlabels train_mosafinale17classes_Labels48.tif uint8".

And  when I run otbcli_TensorflowModelTrain \ with -training.source1.patchsizex 48 \
-training.source1.patchsizey 48 \
I have this error message :

"wModelTrain: itk::ERROR: TensorflowMultisourceModelTrain(0x564fdea48c70): Can't run the tensorflow session !
Tensorflow error message:
Invalid argument: logits and labels must have the same first dimension, got logits shape [8100,18] and labels shape [100]
         [[{{node sparse_softmax_cross_entropy_loss/xentropy/xentropy}}]]
OTB Filter debug message:
Output image buffered region: ImageRegion (0x7ffe967afb80)
  Dimension: 2
  Index: [0, 0]
  Size: [1, 1]

Input #0:
Requested region: ImageRegion (0x7ffe967afbb0)
  Dimension: 2
  Index: [0, 0]
  Size: [48, 4270176]

Tensor shape ("x": {100, 48, 48, 11}
Input #1:
Requested region: ImageRegion (0x7ffe967afbb0)
  Dimension: 2
  Index: [0, 0]
  Size: [1, 88962]

Tensor shape ("y": {100, 1, 1, 1}

So  my question : Do I have any parameters to modify in the python script? 

Thanks for your help

**Comment by remicres** [comment]
Hi @viennois 
This message comes from tensorflow.
It says that the number of labels is wrong.
Here I see 8100 values (instead of 100) which is 81 times greater than it should.
Maybe you have 9x9 labels outputs, instead of 1x1.
You have either to shrink this output or add convolutions to match 1x1

**Comment by viennois** [comment]
Bonjour Rémy
je me permets de te répondre par email car ce sera plus facile pour échanger. Je travaille au laboratoire AMAP sur Montpellier.
Je suis novice en deep learning et j'ai testé ton application OTBTF sur des images sentinel 2 en Afrique Centrale qui donne de bons résultats.
J'aurais souhaiter changer la taille d'échelle d'analyses pour voir l'impact sur les classifications.
Pour cela, je veux pouvoir changer la taille des patchs. J'ai voulu tester une "fenetre d'analyse plus grande" en passant de 16x16 à 48x48 .
Je ne comprends pas trop ta réponse sur 8100 values instead of 100 et je ne sais pas comment augmenter les convolutions sur le script pour matcher à 1x1
Etant sur Montpellier tous les 2, pourrait on se rencontrer pour que tu m'expliques comment modifier le script ? 
Bonne journée
Gaëlle Viennois
UMR AMAP
Ingénieure de Recherche SIG & Télédétection
https://amap.cirad.fr



----- Mail original -----
De: "Rémi Cresson" ***@***.***>
À: "remicres/otbtf" ***@***.***>
Cc: "viennois" ***@***.***>, "Mention" ***@***.***>
Envoyé: Lundi 22 Août 2022 11:44:10
Objet: Re: [remicres/otbtf] Error TensorflowModelTrain with path of dimension 48 instead of 16 (Issue #88)

Hi @viennois 
This message comes from tensorflow.
It says that the number of labels is wrong.
Here I see 8100 values (instead of 100) which is 81 times greater than it should.
Maybe you have 9x9 labels outputs, instead of 1x1.
You have either to shrink this output or add convolutions to match 1x1

**Comment by viennois** [comment]
I have 18 classes and i try to use patches of 48*48.
I 'm just user of our application and I don't know how to use tensorflow
How shrink the output or add convolutions to match 1x1? 
Thanks for your help
